### PR TITLE
Add new Zuul filters endpoint and refactor routes endpoint

### PR DESCRIPTION
**Pull request by darylrobbins**
Added a new filters endpoint to list the current Zuul filters that are in effect. It returns a map keyed by filter type, containing lists of the corresponding filters.

- Create a new endpoints package to house endpoints and moved routes endpoint
- Added a test for the filters endpoint and backfilled tests for the routes endpoint
- Updated documentation to describe new filters endpoint

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@darylrobbins Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1654)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1654) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@darylrobbins Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1654)!

**Comment by darylrobbins** [comment]
Not quite sure what is causing the test failures, but I don't think it's my changes. I saw those same failures when I first ran a build against master (before making any changes). I re-ran the same build without making any changes and it worked.

The test run works successfully on my machine for my changes. Looks like it could be a timing/state propagation issue with those tests?

**Comment by tmack8001** [comment]
Yeah there have been some flaky tests on master recently I've run into as well.

**Comment by darylrobbins** [comment]
As an aside, any particular reason the routes endpoint wasn't added to ZuulServer as well as ZuulProxy? You will always have routes and filters regardless of whether you add the extra Proxy glue.

**Comment by tmack8001** [comment]
@EnableZuulProxy is used when you want to have a zuul setup with the reverse proxy filters
@EnableZuulServer is used to setup base vanilla zuul without any of the proxying filters that were added to ZuulProxy.

Details here https://github.com/spring-cloud/spring-cloud-netflix/issues/104

**Comment by darylrobbins** [comment]
Understood, but why wouldn't I want to have both these endpoints in either case? I was more questioning why the routes endpoint was only enabled in the proxy case.

**Comment by spencergibb** [comment]
Probably an oversight

**Comment by spencergibb** [comment]
Rerunning build

**Comment by ryanjbaxter** [comment]
@darylrobbins looks like some tests are failing, can you look into them?

**Comment by darylrobbins** [comment]
Not sure this is related to my changes, but I'll try to take a look when I have a chance.

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/endpoints/FiltersEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import java.util.*;
+
+/**
+ * Endpoint for listing Zuul filters
+ */
+@ManagedResource(description = "List Zuul filters")
+public class FiltersEndpoint implements MvcEndpoint {
+
+    @RequestMapping(method = RequestMethod.GET)
```
Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/endpoints/FiltersEndpoint.java]: We use tabs instead of spaces.

**Comment by spencergibb** [comment]
ping @darylrobbins this will need a rebase.

**Comment by spencergibb** [comment]
Superceded by #2244

**Pull request closed without merging**